apps/discussions/contracts/DiscussionApp.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "@aragon/os/contracts/common/IForwarder.sol";
+
+
+contract DiscussionApp is IForwarder, AragonApp {
+    using SafeMath for uint256;
+
+    event Post(address indexed author, string postCid, uint discussionThreadId, uint postId, uint createdAt);
+    event Revise(address indexed author, string revisedPostCid, uint discussionThreadId, uint postId, uint createdAt, uint revisedAt);
+    event Hide(address indexed author, uint discussionThreadId, uint postId, uint hiddenAt);
+    event CreateDiscussionThread(uint actionId, bytes _evmScript);
+
+    bytes32 constant public DISCUSSION_POSTER_ROLE = keccak256("DISCUSSION_POSTER_ROLE");
+    string private constant ERROR_CAN_NOT_FORWARD = "DISCUSSIONS_CAN_NOT_FORWARD";
+
+    struct DiscussionPost {
+        address author;
+        string postCid;
+        uint discussionThreadId;
+        uint id;
```

Comment:
I would specify unit256 for all params, and for indexes or anything else that's only being modified based on incrementation you can probably get away with a uint64.